binderhub/builder.py

```diff
@@ -542,31 +590,33 @@ def _check_result(future):
                         BUILD_COUNT.labels(
                             status="failure", **self.repo_metric_labels
                         ).inc()
-
                 await self.emit(event)
 
-        # Launch after building an image
-        if not failed:
-            BUILD_TIME.labels(status="success").observe(
-                time.perf_counter() - build_starttime
-            )
-            BUILD_COUNT.labels(status="success", **self.repo_metric_labels).inc()
-            with LAUNCHES_INPROGRESS.track_inprogress():
-                await self.launch(provider)
-            self.event_log.emit(
-                "binderhub.jupyter.org/launch",
-                5,
-                {
-                    "provider": provider.name,
-                    "spec": spec,
-                    "ref": ref,
-                    "status": "success",
-                    "build_token": self._have_build_token,
-                    "origin": self.settings["normalized_origin"]
-                    if self.settings["normalized_origin"]
-                    else self.request.host,
-                },
-            )
+        if build_only_outcome:
+            return
+        else:
+            # Launch after building an image
+            if not failed:
+                BUILD_TIME.labels(status="success").observe(
+                    time.perf_counter() - build_starttime
+                )
+                BUILD_COUNT.labels(status="success", **self.repo_metric_labels).inc()
```

Comment:
I think we should have BUILD_TIME and BUILD_COUNT tracked before returning independently `if build_only_outcome`.

Let's relocate this part to be under `elif progress.payload == ProgressEvent.BuildStatus.BUILT:` (similarly to how BUILD_TIME is updated if failure is observed on line 587).